swan-lake/development-tutorials/test-document-the-code/test-ballerina-code/mocking.md

```diff
@@ -211,6 +207,191 @@ public function testGetRandomJoke() {
+    test:assertEquals(sendNotification(emailIds), ());
+}
+```
+
+#### Stub resource methods
+
+Similar to remote methods, resource methods also can be stubbed to return a specific value, a series of values or, to do nothing. To mock a resource method, the resource method name(`get`, `post`, `put`, ...etc) and resource path should be specified. Each path parameter in the resource path should be indicated with a colon(`:`) prefix. Similarly, Rest parameters should be indicated with a double colon(`::`) prefix.
+
+***Example:***
+
+Let's consider the following client example.
+
+***main.bal***
+
+```ballerina
+public type Employee record {|
+    readonly string id;
+    string firstName;
+    string lastName;
+    string gender;
+|};
+
+EmpClient empClient = new ();
+
+public client class EmpClient {
+    map<Employee> employees;
+
+    function init() {
+        self.employees = {};
+    }
+
+    resource function get employees/[string... args]() returns string {
+        return "Welcome to the Employees API";
+    }
+
+    resource function get employee/[string id]() returns Employee? {
+        return self.employees[id];
+    }
+
+    resource function get employee/welcome/[string id](string firstName, string lastName) returns string {
+        return "Welcome " + firstName + " " + lastName + ". your ID is " + id;
+    }
+}
+```
+
+##### Stub to return a specific value with a resource method
+
+***main_test.bal***
+ 
+This test stubs the behavior of the below resource method to return a specific value in 4 ways:
+
+```ballerina
+    resource function get employee/welcome/[string id](string firstName, string lastName) returns string {
+        return "Welcome " + firstName + " " + lastName + ". your ID is " + id;
+    }
+```
+
+1. Stubbing to return a specific value in general
+2. Stubbing to return a specific value based on the path parameter
+3. Stubbing to return a specific value based on the method arguments
+4. Stubbing to return a specific value based on the path parameter and the method arguments
+
+```ballerina
+@test:Config {}
+function testWelcomeEmployee() {
+    empClient = test:mock(EmpClient);
+    // Stubbing to return a specific value in general
+    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get").thenReturn("Stub_1");
+
+    // Stubbing to return a specific value on a specific path parameter
+    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get").withPathParameters({id: ""}).thenReturn("Stub_2");
+
+    // Stubbing to return a specific value on a specific method arguments
+    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get").withArguments("", "").thenReturn("Stub_3");
+
+    // Stubbing to return a specific value on a specific path parameter and method arguments
+    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get").withPathParameters({id: ""}).withArguments("", "").thenReturn("Stub_4");
```

Comment:
```suggestion
    test:prepare(empClient).whenResource("employee/welcome/:id").onMethod("get")
    .withPathParameters({id: ""}).withArguments("", "").thenReturn("Stub_4");
```

Reply:
This is shown properly in the website view as this is a code block.